Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -490,6 +490,21 @@ static int RunGpuOnCpu(int ticks)
   return -available_ticks + GPU_TIME_SLOT_SIZE;
 }
 
+void SyncGPUForRegisterAccess()
+{
+  if (SConfig::GetInstance().bCPUThread && !s_use_deterministic_gpu_thread)
+  {
+    // Dual core - kick the GPU, wait for completion.
+    RunGpu();
+    s_gpu_mainloop.Wait();
```

Comment:
This is a bit gross.
RunGpu() shall only be called on queuing new work for the GPU thread, or if you explicitly want the thread to wake up.
Also if the game idles in such an infinity loop, this will run terrible slow.
What do you think about calling WaitForGpuThread() instead, maybe only if SyncGpu is enabled?

But I see how such a hack may improve the stability. This tries to emulate an infinite fast GPU by blocking on the first state lookup. *But* please drop the RunGpu() command.